src/main/kotlin/com/learnspigot/bot/command/StartVoteCommand.kt

```diff
@@ -1,110 +1,55 @@
-import com.learnspigot.bot.LearnSpigotBot.Companion.findUserProfile
 import com.learnspigot.bot.LearnSpigotBot.Companion.replyEmbed
-import com.learnspigot.bot.entity.UserProfile
-import com.learnspigot.bot.manager.LeaderboardManager
-import com.learnspigot.bot.util.KnowledgeBaseTypes
+import com.learnspigot.bot.manager.KnowledgeBaseManager
+import com.learnspigot.bot.util.KnowledgeBaseType
 import dev.minn.jda.ktx.events.onCommand
 import dev.minn.jda.ktx.interactions.commands.restrict
 import dev.minn.jda.ktx.interactions.commands.upsertCommand
-import dev.minn.jda.ktx.messages.Embed
-import dev.morphia.Datastore
 import net.dv8tion.jda.api.JDA
+import net.dv8tion.jda.api.Permission
 import net.dv8tion.jda.api.entities.Guild
-import net.dv8tion.jda.api.entities.Message
-import net.dv8tion.jda.api.entities.MessageEmbed
-import net.dv8tion.jda.api.entities.channel.forums.ForumTag
-import net.dv8tion.jda.api.entities.emoji.Emoji
-import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent
-import java.util.*
-import java.util.concurrent.TimeUnit
-import kotlin.concurrent.schedule
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel
 
 class StartVoteCommand(
     private val guild: Guild,
     private val bot: JDA,
-    private val datastore: Datastore,
-    private val leaderboardManager: LeaderboardManager
+    private val knowledgeBaseManager: KnowledgeBaseManager
 ) {
     fun startVoteCommand() {
         guild.upsertCommand("startvote", "Start a vote for project or tutorial") {
-            restrict(guild = true)
+            restrict(guild = true, Permission.ADMINISTRATOR)
             bot.onCommand("startvote") {
-                if (guild.getMemberById(it.user.id)!!.roles.contains(guild.getRoleById(System.getenv("SUPPORT_ROLE_ID"))) && bot.getThreadChannelById(
-                        it.channel!!.id
-                    )?.parentChannel?.id != null && bot.getThreadChannelById(it.channel!!.id)!!.parentChannel.id == System.getenv(
-                        "FOR_REVIEW_CHANNEL_ID"
-                    )
-                ) {
-                    it.replyEmbed({
-                        title = "Success"
-                        description = "A vote has been started, results will be announced in 24 hours"
-                        color = EMBED_COLOR
-                    }).queue()
-                    val tutorialEmojis = mutableListOf("0️⃣", "1️⃣", "2️⃣", "3️⃣", "4️⃣", "5️⃣")
-                    val projectEmojis =
-                        mutableListOf("0️⃣", "1️⃣", "2️⃣", "3️⃣", "4️⃣", "5️⃣", "6️⃣", "7️⃣", "8️⃣", "9️⃣", "🔟")
-                    val type = KnowledgeBaseTypes.valueOf(bot.getThreadChannelById(it.channel!!.id)!!.appliedTags.find { tag -> tag.name == "Tutorial" || tag.name == "Project" }?.name!!.uppercase())
+                if(!it.member!!.roles.contains(guild.getRoleById(System.getenv("SUPPORT_ROLE_ID")))) return@onCommand
```

Comment:
is it actually admin must have support role in the server?
if yes it is quite redundant, as not many people have that permission

Reply:
We can fine tune the permissions in discord